packages/serverpod_client/lib/src/serverpod_client_shared.dart

```diff
@@ -427,6 +429,54 @@ abstract class ServerpodClientShared extends EndpointCaller {
     await _sendControlCommandToStream('auth', {'key': authKey});
   }
 
+  /// Performs the actual call to the server.
+  /// Subclasses implement bespoke logic for e.g. web / io clients.
+  Future<String> callServerEndpointImpl<T>(
+    Uri url, {
+    required String body,
+  });
+
+  @override
+  Future<T> callServerEndpoint<T>(
+      String endpoint, String method, Map<String, dynamic> args) async {
+    var callContext = MethodCallContext(
+      endpointName: endpoint,
+      methodName: method,
+      arguments: args,
+    );
+
+    try {
+      var body =
+          formatArgs(args, await authenticationKeyManager?.get(), method);
+      var url = Uri.parse('$host$endpoint');
+
+      var data = await callServerEndpointImpl(
+        url,
+        body: body,
+      );
+
+      T result;
+      if (T == getType<void>()) {
+        result = returnVoid() as T;
+      } else {
+        result = parseData<T>(data, T, serializationManager);
+      }
+
+      onSucceededCall?.call(callContext);
+      return result;
+    } catch (e, s) {
+      onFailedCall?.call(callContext, e, s);
+
+      if (logFailedCalls) {
+        // ignore: avoid_print
```

Comment:
This code is moved from the subclasses. Should we perhaps replace the print() calls with a logger?